staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -404,6 +418,16 @@ func (f *DeltaFIFO) Pop(process PopProcessFunc) (interface{}, error) {
 	defer f.lock.Unlock()
 	for {
 		for len(f.queue) == 0 {
+			// When the queue is empty, invocation of Pop() is blocked until new item is enqueued.
+			// When Close() is called, the f.closed is set and the condition is broadcasted.
+			// Which causes this loop to continue and return from the Pop().
+			f.closedLock.Lock()
```

Comment:
nit: could this be pulled into a method for `IsClosed` so I don't have to pair locks when I'm reading this?